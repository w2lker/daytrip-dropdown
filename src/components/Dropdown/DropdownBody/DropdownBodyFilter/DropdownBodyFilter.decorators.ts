import withStyles from "react-jss";

import DropdownBodyFilter from './DropdownBodyFilter';
import dropdownBodyFilterStyles from './DropdownBodyFilter.styles';

const styledComponent = withStyles(dropdownBodyFilterStyles)(DropdownBodyFilter);

export default styledComponent;
