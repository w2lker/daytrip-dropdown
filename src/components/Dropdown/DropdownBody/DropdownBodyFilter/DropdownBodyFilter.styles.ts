const dropdownBodyFilterStyles = {
  wrapper: {
    padding: 0,
  },
  magnify: {

  },
  input: {
    padding: 0,
  },
  divider: {

  },
};

export default dropdownBodyFilterStyles;
