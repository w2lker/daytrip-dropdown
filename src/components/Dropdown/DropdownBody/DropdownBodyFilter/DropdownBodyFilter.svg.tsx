import React, {ReactElement} from "react";

export const magnify = (color: string = '#299BF6'): ReactElement => {
  return (
    <svg width="14px" height="14px" viewBox="0 0 14 14">
        <path d="M5.2,0 C8.072,0 10.4,2.328 10.4,5.2 C10.4,6.488 9.928,7.672 9.152,8.584 L9.368,8.8 L10,8.8 L14,12.8
        L12.8,14 L8.8,10 L8.8,9.368 L8.584,9.152 C7.672,9.928 6.488,10.4 5.2,10.4 C2.328,10.4 0,8.072 0,5.2 C0,2.328
        2.328,0 5.2,0 Z M5.2,1.6 C3.2,1.6 1.6,3.2 1.6,5.2 C1.6,7.2 3.2,8.8 5.2,8.8 C7.2,8.8 8.8,7.2 8.8,5.2 C8.8,3.2
        7.2,1.6 5.2,1.6 Z" fill={color}/>
    </svg>
  );
};
