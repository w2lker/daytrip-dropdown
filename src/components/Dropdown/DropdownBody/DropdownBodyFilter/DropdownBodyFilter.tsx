import React, {useEffect, useRef} from "react";

interface IDropdownBodyFilterProps {
    value: string;
    onChange: (val: string) => void;
    classes: any;
    // TODO: add on select previous, on select next and on enter keypress
}

const DropdownBodyFilter: React.FC<IDropdownBodyFilterProps> = (props) => {
    const { value, onChange } = props;
    const inputRef = useRef(null);
    useEffect(() => {
        // @ts-ignore
        inputRef && inputRef.current && inputRef.current.focus && inputRef.current.focus();
    });
    const handleFilterEnter = (event: React.FormEvent<HTMLInputElement>):void => onChange((event.target as HTMLInputElement).value);
    return (
        <div>
        {/*  TODO: add search icon here  */}
            <input
                ref={inputRef}
                value={value}
                onChange={handleFilterEnter}
            />
        </div>
    )
};

export default DropdownBodyFilter;
