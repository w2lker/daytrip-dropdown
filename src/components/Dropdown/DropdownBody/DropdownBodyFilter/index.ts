import DropdownBodyFilter from './DropdownBodyFilter.decorators';

export default DropdownBodyFilter;
