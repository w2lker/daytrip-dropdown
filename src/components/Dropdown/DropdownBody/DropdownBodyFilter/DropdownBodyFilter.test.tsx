import React from 'react';

import { mount, shallow } from 'enzyme';

import DropdownBodyFilter, { IDropdownBodyFilterProps } from './DropdownBodyFilter';
import DropdownBodyFilterDecorated from './DropdownBodyFilter.decorators';
import dropdownBodyFilterStyles from './DropdownBodyFilter.styles';


const setup = () => {
  const classes = {
    wrapper: 'dropdownBodyFilter-wrapper',
    magnify: 'dropdownBodyFilter-magnify',
    input: 'dropdownBodyFilter-input',
    divider: 'dropdownBodyFilter-divider',
  };

  const classesKeys = Object.keys(classes);
  const sampleProps: IDropdownBodyFilterProps = {
    value: 'some filter',
    onClose: jest.fn(),
    onFilterChange: jest.fn(),
    onSelectPerform: jest.fn(),
    onSelectNext: jest.fn(),
    onSelectPrev: jest.fn(),
    classes,
  };

  return {classes, classesKeys, sampleProps}
};

describe('DropdownBodyFilter component', () => {

  it('match snapshot', () => {
    const { sampleProps } = setup();
    const component = shallow(<DropdownBodyFilter {...sampleProps} />);
    expect(component.debug()).toMatchSnapshot();
  });

  it('renders input', () => {
      const { classes, sampleProps } = setup();
    const component = shallow(<DropdownBodyFilter {...sampleProps} />);
    expect(component.find(`.${classes.input}`).length).toBe(1);
  });

  it('perform change', () => {
    const { classes, sampleProps } = setup();
    const component = shallow(<DropdownBodyFilter {...sampleProps} />);
    const newFilter = 'new filter';
    component.find(`.${classes.input}`).simulate('change', { target: { value: newFilter } });
    expect(sampleProps.onFilterChange).toBeCalledTimes(1);
    // @ts-ignore
    const passedArg = sampleProps.onFilterChange.mock.calls[0][0];
    expect(passedArg).toBe(newFilter);
  });

  it('handle switch selections', () => {
      const { classes, sampleProps } = setup();
      const preventFunction = jest.fn();
      const keyPressParams = (keyName: string, shiftKey?: boolean) => ({
        key: keyName,
        shiftKey: shiftKey,
        preventDefault: preventFunction,
      });
      const component = shallow(<DropdownBodyFilter {...sampleProps} />);
      const input = component.find(`.${classes.input}`);
      input.simulate('keyDown', keyPressParams('Enter'));
      expect(sampleProps.onSelectPerform).toBeCalledTimes(1);
      input.simulate('keyDown', keyPressParams('ArrowUp'));
      expect(sampleProps.onSelectPrev).toBeCalledTimes(1);
      input.simulate('keyDown', keyPressParams('ArrowDown'));
      expect(sampleProps.onSelectNext).toBeCalledTimes(1);
      input.simulate('keyDown', keyPressParams('Tab', true));
      expect(sampleProps.onSelectPrev).toBeCalledTimes(2);
      input.simulate('keyDown', keyPressParams('Tab'));
      expect(sampleProps.onSelectNext).toBeCalledTimes(2);
      expect(preventFunction).toBeCalledTimes(4);
  });

  it('handle close on input blur', () => {
    const { classes, sampleProps } = setup();
    const component = shallow(<DropdownBodyFilter {...sampleProps} />);
    const input = component.find(`.${classes.input}`);
    input.simulate('blur');
    expect(sampleProps.onClose).toBeCalledTimes(1);
  });

  it('handle close on Esc keypress', () => {
    const { classes, sampleProps } = setup();
    const component = shallow(<DropdownBodyFilter {...sampleProps} />);
    const input = component.find(`.${classes.input}`);
    input.simulate('keyDown', {key: 'Escape'});
    expect(sampleProps.onClose).toBeCalledTimes(1);
  })
});

describe('DropdownBodyFilter styles', () => {
  it('contains principle classes', () => {
    const { classesKeys } = setup();
    expect(dropdownBodyFilterStyles).toBeDefined();
    classesKeys.forEach((keyValue) => {
      // @ts-ignore
      expect(dropdownBodyFilterStyles[keyValue]).toBeDefined();
    });
  });
});

describe('DropdownBodyFilter decorators', () => {
  it('provides styled classes from decorators', () => {
    const { classesKeys, sampleProps } = setup();
    const component = mount(<DropdownBodyFilterDecorated {...sampleProps} />);
    // @ts-ignore
    const assignedClasses = component.find('DropdownBodyFilter').props().classes;
    classesKeys.forEach(keyValue => {
      expect(assignedClasses[keyValue]).toBeDefined();
    });
  });
});
