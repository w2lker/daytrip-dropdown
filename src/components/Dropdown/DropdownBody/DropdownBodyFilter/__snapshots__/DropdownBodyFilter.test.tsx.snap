// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownBodyFilter component match snapshot 1`] = `
"<div className=\\"dropdownBodyFilter-wrapper\\">
  <input className=\\"dropdownBodyFilter-input\\" value=\\"some filter\\" onChange={[Function: handleFilterChange]} onKeyPress={[Function: handleFilterKeyPress]} onBlur={[undefined]} />
</div>"
`;
